django/db/backends/mysql/schema.py

```diff
@@ -69,6 +70,12 @@ def _field_should_be_indexed(self, model, field):
                 field.get_internal_type() == 'ForeignKey' and
                 field.db_constraint):
             return False
+
+        # MySQL's blob/text/json fields do not support indexing of whole field
```

Comment:
How about checking for `field.get_internal_type()` to be `text`, etc and bail out with `False`. I do not understand how `skip_default()` relates to an index being created or not.